mlflow/tracking/client.py

```diff
@@ -469,6 +471,43 @@ def search_registered_models(
         :return: A PagedList of :py:class:`mlflow.entities.model_registry.RegisteredModel` objects
                 that satisfy the search expressions. The pagination token for the next page can be
                 obtained via the ``token`` attribute of the object.
+
+        .. code-block:: python
+            :caption: Example
+
+            import mlflow
+
+            model_name="CordobaWeatherForecastModel"
+            filter = "name='{}'".format(model_name)
+            client = mlflow.tracking.MlflowClient()
+
+            # Get search results filtered by the registered model name
+            results = client.search_registered_models(filter_string=filter)
+            print("-" * 80)
+            for res in results:
+                for mv in res.latest_versions:
+                    print("name={}; run_id={}; version={}".format(mv.name, mv.run_id, mv.version))
+            # Get all registered models and order them by ascending order of the names
+            results = client.search_registered_models(order_by=["name ASC"])
+            print("-" * 80)
+            for res in results:
+                for mv in res.latest_versions:
+                    print("name={}; run_id={}; version={}".format(mv.name, mv.run_id, mv.version))
```

Comment:
Should we add examples for `LIKE` and `ILIKE` operators?

Reply:
I'm going to do that with `search_runs()` as part of another PR that covers fluent `mflow` APIs. [Search runs](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.search_runs) will have these `LIKE`and `ILIKE` operators.  For this issue, I quickly wanted to address it. 

Both these APIs are experimental; we might also add 'tag' too.

Reply:
> I belive these APIs, for now, only support name= operators. If you use LIKE such as query = "name LIKE '%Azure%'", you'll get an exception.

I think `search_registered_models` supports `LIKE` and `ILIKE` operators. 

```python
import mlflow

# mkdir -p /tmp/mlflow/
mlflow.set_tracking_uri("sqlite:////tmp/mlflow/db")


class Model(mlflow.pyfunc.PythonModel):
    pass


mlflow.pyfunc.log_model("model", python_model=Model(), registered_model_name="model")
client = mlflow.tracking.MlflowClient()
print(client.search_registered_models("name LIKE 'mode%'"))
```

`search_model_versions` doesn't though: https://github.com/mlflow/mlflow/search?q=%22only+supports+equality%22&unscoped_q=%22only+supports+equality%22

Reply:
`search_model_versions` doesn't:

https://github.com/mlflow/mlflow/blob/3936705d9a41ada8e526c7981bb0a119862d53c3/mlflow/store/model_registry/sqlalchemy_store.py#L748-L752

Reply:
Excellent point @harupy! Let me add that operator to `search_registered_models`. I think it's good to put that as en example
in the code snippet for `search_registered_models`.

Reply:
Done!